### PR TITLE
Arthropod enhancement: Dragonflies

### PR DESCRIPTION
**Pull request by Venera3**
#### Summary

SUMMARY: Content "Arthropod enhancement: Dragonflies"

#### Purpose of change

Make dragonflies almost as cool as they are in real life.

#### Describe the solution

IRL dragonflies have a pretty interesting life cycle, beginning as strictly waterbound predators with honest to god Alien-jaws, and then molting on land into aerial killing machines so specialised that they can't even walk properly anymore. In Cata they got the standard "blobbed insect" evolution line of small and boring -> big and boring -> fuckhuge and boring, and that just can't stay like that.
So I:
-Added naiads, the aquatic larvae with a custom monfac
-Redid vanilla dragonflies to be threatening, the mega variant to be a mating pair that fused together and lays eggs
-Made them hunt (if they get something to hunt, see below)
-While I was there I added a harvest entry for flying critters and added mosquitos, flies, dermatiks, and locusts there

#### Describe alternatives you've considered

Leave out the naiads, because they do cause some weird problems (the eggs need to rot in water for an aquatic critter to spawn), but they are weird and cool and besides the rivers are too safe as is. 

Not give the giant dragonflies grabbing attacks or not having them swarm, but grabbing makes sense as a special for them and swarming lets them get more mileage out of it. Decided against group morale to keep them from ganking too much and to hammer home that these are only opportunistic clusters, not social packs (think sharks, not dolphins).

Keep the mega variant as flying, but raise its speed so it stays halfway dangerous, but since that thing weighs almost a ton I would have needed to add chitinous jet engines with it. The suspension of disbelief is pretty fragile as is with the big fliers, hydrogen bubbles notwithstanding.

#### Testing

All variants spawn in the desired groups and biomes, and hunt their designated prey groups. 
Eggs that rot in water spawn naiads with a higher probability in the summer, in or out of the reality bubble. 
An egg rotting on land causes no errors. 
Butchering give the new results and message.
Combat tested naiads and dragonflies pretty extensively, fused variant still needs a bit of work.

#### Additional context

Dragonfly naiads are smallish aquatic critters with a reach attack to simulate their snapping lower jaw, with their own aquatic predator monster faction to annoy the rest of the fish.  Generally more annoying than dangerous, but they should spice up your fishing trips/canoe rides. They spawn in the river and fish mongroup in packs of 1-3 and grow into the giant variant.

Made the vanilla "giant" variant to see better, added a grab attack cause those legs can't do anything else, upped damage, and played with their AI so that if they decide to get a taste of your face they don't get scared before taking a bite. Dangerous for Day 1 survivors or if you're already having a bad day, but they have pretty low morale so can be convinced to leave you alone with a lucky hit or two. From around Summer on they'll start spawning in swarms, so that should keep the challenge up. They also hunt some monfactions that made sense.

The mega variant is/are now the fused dragonflies, a huge, slow-ish (but still faster than they were), nonflying egglayer that probably can't catch you, but if it does you'll have a bad time. Eggs are on a two-day rot timer and spawn naiads. Sadly I could not see a way to get them to spend more time in water, so most of the eggs will be duds. They also have their own harvest entry with a custom message.

Giants can spawn from day one in swamps, get increasing freq and packsize till day 42+, from then on fused dragonflies can also spawn with a low freq.

Since the swamps are pretty deserted at the moment (owing to the spawn probability calculations), you can't really see them hunt. A wholesale overhaul of the swamp spawn list is out of scope for this PR, but I took the liberty to give dragonflies a relatively high chance to spawn from the "midgame" on. 

I tried to simulate a life cycle, so I replaced all evolutions with "age_grows", which keeps the eggs from spawning adults after x game days (since apperently evolution happens to freshly hatched critters, too). That has the added bonus of still meeting the more dangerous giant dragonflies in the late game.

This is my second PR, so any feedback is welcome. I'm especially looking for alternative names for the fused dragonflies, since that does not play well with the combat log, and general tonal feedback for the descriptions/messages. 

Dragonfly Naiad:
https://www.uky.edu/Ag/CritterFiles/casefile/insects/dragonflies/dragnaiad.jpg
Mating pair:
https://i.dailymail.co.uk/i/pix/2011/07/28/article-2019723-0D3233F500000578-121_634x755.jpg

### CODE REVIEW

**Comment by Venera3** [comment]
Combat balance:
Since the monster melee system is kind of half-documented (well, 3/4) this is all brute-force testing and eyeballing.
Testing done on a day1 char (0/0/0 skills with a steak knife), early midgame (3/3/3, aluminum bat or quarterstaff, no armor ), and lategame ( 7/7/7 Bionic assasin, monomolecular blade or nodachi, all reinforced survivor gear). Each got 10 tries, no kiting, no reach attacks, no ranged, brawling as a MA, and basically getting surrounded and TABbing.

Naiads can be pretty easily dispatched, and since they are only dangerous in a thin strip on the shore can't really mess you up. On a boat they are a slightly bigger problem, but by the time you build a boat you should be more than well enough equipped to deal with them. They also can't see that good.

A single adult can be made to flee by a day1  2 times out of 10, will be killed 10/10 by an early-midgame char (5x without getting hit), and pose no threat whatever in the lategame.
A pack of 3 (spawning from week one) will dispatch a day1 all the time, a midgame char has 7/10 chance to make the pack flee with a bat and 10/10 with a quarterstaff (those sweep attacks are improtant), and mildly annoy a lategame char.
A pack of 7 (day 42+) leaves a quarterstaff user with a 1/10 chance to walk crawl away almost dead and will be killed by a lategame char 10/10 with varying damage (highest was 50% and a few ripped clothes). A sweep attack makes a big difference here too.

The fused pair shoud be outrun/kited till lategame, there a single one can do 20%dmg on average, a pack of 5 will mess up even a late game character.

Since the adultss only spawn in swamps and need 24 days to mature form the naiads in rivers/ponds I think thats about reasonable, but feedback is welcome.

They are all semi-natural animals, so they actually flee at trouble, and I gave the adults and fused variants pretty high AP damage to keep a bit of challenge without screwing over the earlygame balance too much. The grabs on the adult have a long countdown to technically let you flee, but since they are faster than you that's not generally a good idea.

Still doing some reproduction/spawning testing, then this should be done.

**Comment by Venera3** [comment]
All right, everything seems to work as advertised (they spawn, grow up, reproduce, hunt other insects).

**Comment by Venera3** [comment]
Since I'm looking to make flying insects more common in swamps, they can stand to give less chitin/meat

**Comment by Venera3** [comment]
Okay, I figured out the butchery snippets and added buzzzing to all flying insects to make them even more annoying. Promise I'm done fiddling now.

File: data/json/snippets/mutant_anatomy.json [hunk 1]
```diff
@@ -146,7 +149,40 @@
     "text": [
       "The meat inside gives off a horrifying stench, and seems to be covered in thin damp hair, like a newborn animal gone horribly wrong",
       "Small bundles of fiber break loose as you work, splitting open to reveal twisted, half-formed copies of the creature itself",
-      "It is difficult to work, as the spongey tissue tears apart under your tools"
+      "It is difficult to work, as the spongey tissue tears apart under your tools",
+      "Despite its outward appearance, both the organs and the endlymph of this creature is pristinely white",
```
Comment (I-am-Erk) [inline data/json/snippets/mutant_anatomy.json:153]:
```suggestion
      "Despite its outward appearance, both the organs and the endolymph of this creature is pristinely white",
```

**Comment by I-am-Erk** [comment]
Ready to merge once typo fixed and conflict resolved

**Comment by Venera3** [comment]
Updated them to the universal scaling times for the swamps, also forgot to actually add the loudmoves flag to the rest.

**Comment by AngularFruit** [comment]
> #### Summary
> SUMMARY: Content "Arthropod enhancement: Dragonflies"
> 
> #### Purpose of change
> Make dragonflies almost as cool as they are in real life.
> 
> #### Describe the solution
> IRL dragonflies have a pretty interesting life cycle, beginning as strictly waterbound predators with honest to god Alien-jaws, and then molting on land into aerial killing machines so specialised that they can't even walk properly anymore. In Cata they got the standard "blobbed insect" evolution line of small and boring -> big and boring -> fuckhuge and boring, and that just can't stay like that.
> So I:
> -Added naiads, the aquatic larvae with a custom monfac
> -Redid vanilla dragonflies to be threatening, the mega variant to be a mating pair that fused together and lays eggs
> -Made them hunt (if they get something to hunt, see below)
> -While I was there I added a harvest entry for flying critters and added mosquitos, flies, dermatiks, and locusts there
> 
> #### Describe alternatives you've considered
> Leave out the naiads, because they do cause some weird problems (the eggs need to rot in water for an aquatic critter to spawn), but they are weird and cool and besides the rivers are too safe as is.
> 
> Not give the giant dragonflies grabbing attacks or not having them swarm, but grabbing makes sense as a special for them and swarming lets them get more mileage out of it. Decided against group morale to keep them from ganking too much and to hammer home that these are only opportunistic clusters, not social packs (think sharks, not dolphins).
> 
> Keep the mega variant as flying, but raise its speed so it stays halfway dangerous, but since that thing weighs almost a ton I would have needed to add chitinous jet engines with it. The suspension of disbelief is pretty fragile as is with the big fliers, hydrogen bubbles notwithstanding.
> 
> #### Testing
> All variants spawn in the desired groups and biomes, and hunt their designated prey groups.
> Eggs that rot in water spawn naiads with a higher probability in the summer, in or out of the reality bubble.
> An egg rotting on land causes no errors.
> Butchering give the new results and message.
> Combat tested naiads and dragonflies pretty extensively, fused variant still needs a bit of work.
> 
> #### Additional context
> Dragonfly naiads are smallish aquatic critters with a reach attack to simulate their snapping lower jaw, with their own aquatic predator monster faction to annoy the rest of the fish. Generally more annoying than dangerous, but they should spice up your fishing trips/canoe rides. They spawn in the river and fish mongroup in packs of 1-3 and grow into the giant variant.
> 
> Made the vanilla "giant" variant to see better, added a grab attack cause those legs can't do anything else, upped damage, and played with their AI so that if they decide to get a taste of your face they don't get scared before taking a bite. Dangerous for Day 1 survivors or if you're already having a bad day, but they have pretty low morale so can be convinced to leave you alone with a lucky hit or two. From around Summer on they'll start spawning in swarms, so that should keep the challenge up. They also hunt some monfactions that made sense.
> 
> The mega variant is/are now the fused dragonflies, a huge, slow-ish (but still faster than they were), nonflying egglayer that probably can't catch you, but if it does you'll have a bad time. Eggs are on a two-day rot timer and spawn naiads. Sadly I could not see a way to get them to spend more time in water, so most of the eggs will be duds. They also have their own harvest entry with a custom message.
> 
> Giants can spawn from day one in swamps, get increasing freq and packsize till day 42+, from then on fused dragonflies can also spawn with a low freq.
> 
> Since the swamps are pretty deserted at the moment (owing to the spawn probability calculations), you can't really see them hunt. A wholesale overhaul of the swamp spawn list is out of scope for this PR, but I took the liberty to give dragonflies a relatively high chance to spawn from the "midgame" on.
> 
> I tried to simulate a life cycle, so I replaced all evolutions with "age_grows", which keeps the eggs from spawning adults after x game days (since apperently evolution happens to freshly hatched critters, too). That has the added bonus of still meeting the more dangerous giant dragonflies in the late game.
> 
> This is my second PR, so any feedback is welcome. I'm especially looking for alternative names for the fused dragonflies, since that does not play well with the combat log, and general tonal feedback for the descriptions/messages.
> 
> Dragonfly Naiad:
> https://www.uky.edu/Ag/CritterFiles/casefile/insects/dragonflies/dragnaiad.jpg
> Mating pair:
> https://i.dailymail.co.uk/i/pix/2011/07/28/article-2019723-0D3233F500000578-121_634x755.jpg

Not sure if this is the proper place to suggest this, I don't really use github sorry if it isn't. But as for names for the fused dragonfly how about Dyad-fly? or Helix-fly? Chitinous Amalgam? Dragon Flyad? again sorry if this isn't the right place.

**Pull request merged**